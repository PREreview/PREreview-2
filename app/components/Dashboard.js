import React from 'react'
// import styled from 'styled-components'

// const Basic = styled.div`
//   background: blue;
//   // height: 1100px;
//   overflow-y: auto;
// `

import Page from './Page'

const Dashboard = () => (
  <Page>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>

    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
    <h1>The Dashboard!</h1>
  </Page>
)

export default Dashboard
