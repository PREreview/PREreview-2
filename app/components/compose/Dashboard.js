/* eslint-disable react/prop-types */

import React from 'react'
import { Adopt } from 'react-adopt'
import { withApollo } from 'react-apollo'
import { get } from 'lodash'

import {
  createSubmission as createSubmissionMutation,
  createTeam as createTeamMutation,
  deleteArticle,
  getCurrentUser as getCurrentUserQuery,
  getDashboardArticles as getDashboardArticlesQuery,
  getGlobalTeams as getGlobalTeamsQuery,
  handleInvitation,
} from './pieces'

/* eslint-disable sort-keys */
const mapper = {
  getCurrentUserQuery,
  createSubmissionMutation,
  createTeamMutation,
  deleteArticle,
  getDashboardArticlesQuery,
  getGlobalTeamsQuery,
  handleInvitation,
}
/* eslint-enable sort-keys */

const getTeamByType = (teams, type) =>
  teams && teams.find(t => t.teamType === type)

/* eslint-disable-next-line arrow-body-style */
const mapProps = args => {
  // console.log(args.getDashboardArticlesQuery)

  return {
    authorArticles: get(
      args.getDashboardArticlesQuery,
      'data.dashboardArticles.author',
    ),
    createSubmission: args.createSubmissionMutation.createSubmission,
    createTeam: args.createTeamMutation.createTeam,
    currentUser: args.getCurrentUserQuery.data.currentUser,
    deleteArticle: args.deleteArticle.deleteArticle,
    editorArticles: get(
      args.getDashboardArticlesQuery,
      'data.dashboardArticles.editor',
    ),
    globalEditorTeam: getTeamByType(
      args.getGlobalTeamsQuery.data.globalTeams,
      'editors',
    ),
    globalScienceOfficerTeam: getTeamByType(
      args.getGlobalTeamsQuery.data.globalTeams,
      'scienceOfficers',
    ),
    handleInvitation: args.handleInvitation.handleInvitation,
    isGlobal: get(
      args.getDashboardArticlesQuery,
      'data.dashboardArticles.isGlobal',
    ),
    loading:
      args.getGlobalTeamsQuery.loading ||
      args.getDashboardArticlesQuery.loading,
    reviewerArticles: get(
      args.getDashboardArticlesQuery,
      'data.dashboardArticles.reviewer',
    ),
  }
}

const Composed = ({ render, ...props }) => (
  <Adopt mapper={mapper} mapProps={mapProps}>
    {mappedProps => render({ ...props, ...mappedProps })}
  </Adopt>
)

export default withApollo(Composed)
