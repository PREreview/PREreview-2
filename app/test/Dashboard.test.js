import React from 'react'
import { shallow } from 'enzyme'

import Dashboard from '../components/Dashboard'

describe('Dashboard', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Dashboard />)
    expect(wrapper).toMatchSnapshot()
  })
})
