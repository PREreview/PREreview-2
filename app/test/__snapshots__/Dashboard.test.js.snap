// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches the snapshot 1`] = `
<div>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
  <h1>
    The Dashboard!
  </h1>
</div>
`;
